### PR TITLE
CP-29015: Set correct ciphers for stunnel client

### PR DESCRIPTION
**Pull request by xiewei20082008**
Signed-off-by: Kun Ma <kun.ma@citrix.com>

### CODE REVIEW

**Comment by lindig** [comment]
I think asking @johnelse for a review is a mixup. You want @jonludlam. From my perspective, the code looks good.

File: stunnel/stunnel.ml [hunk 1]
```diff
@@ -140,35 +140,40 @@ let config_file verify_cert extended_diagnosis host port legacy =
...
+        ; "options = NO_SSLv2"
+        ; "options = NO_SSLv3"
+        ; "ciphers = " ^ (good_ciphers ()) ^ (match !legacy_ciphersuites with "" -> "" | s -> (":" ^ s))
+        ]
+      else 
+        [ "sslVersion = TLSv1.2"
+        ; "ciphers = " ^ (good_ciphers ())
+        ]
+      ; ["curve = secp384r1"]
+    ]
```
Comment (lindig) [inline stunnel/stunnel.ml]: To make sure the last line has a newline, you could add one empty string as the last element. String.concat only inserts newlines between lines.

```
String.concat "\n" @@ List.concat
[ ...
; [""] (* ensure \n at last line above *)
]
```

File: stunnel/stunnel.ml [hunk 2]
```diff
@@ -140,35 +140,40 @@ let config_file verify_cert extended_diagnosis host port legacy =
...
   in
-  String.concat "" (List.map (fun x -> x ^ "\n") lines)
+  lines
```
[the remark this answers is not in the document]
Comment (xiewei20082008) [reply stunnel/stunnel.ml]: Done.

File: stunnel/stunnel.ml (hunk 1 above)
Comment (xiewei20082008) [reply stunnel/stunnel.ml]: Done.

**Comment by jonludlam** [comment]
Looks good to me.

**Pull request merged**